ocpp_codec/validators.py

```diff
@@ -216,3 +216,30 @@ def to_json(self, value):
                 value=value, enum_class=self.enum_class.__name__,
             )
         return value.value
+
+
+class OutgoingMessageDecimalEncoder(BaseEncoder):
+    """Encoder to limit the precision of decimal values sent to charge points.
+
+    This encoder doesn't restricts the precision of float values received from the charge point, as we *must* keep the
+    full resolution, but limits it to 6 places when sending a value, as required by OCPP 2.0 spec (section 2.1.3).
+    """
+
+    def from_json(self, json_value):
+        return json_value
+
+    def to_json(self, value):
+        try:
+            value = float(value)
+        except ValueError:
+            raise errors.TypeConstraintViolationError(f"Input '{value}' cannot be cast to float", value=value)
+
+        components = str(value).split('.')
+        if len(components) == 2:
+            whole, dec = components
+            dec = dec[:6]
+            value = float(f'{whole}.{dec}')
+        else:
+            # No decimal part
+            value = float(components[0])
+        return value
```

Comment:
To avoid issues (see my comment below), you should rely on `decimal` rather than reinventing the wheel here.